x-pack/solutions/security/plugins/cloud_security_posture/server/routes/graph/parse_records.ts

```diff
@@ -151,31 +179,59 @@ const createNodes = (records: GraphEdge[], context: Omit<ParseContext, 'edgesMap
   }
 };
 
-const determineEntityNodeShape = (
+const determineEntityNodeVisualProps = (
   actorId: string,
   ips: string[],
   hosts: string[],
-  users: string[]
+  users: string[],
+  entitiesData: NodeDocumentDataModel[] = []
 ): {
   shape: EntityNodeDataModel['shape'];
   icon?: string;
 } => {
+  // try to find an exact match by entity id
+  const matchingEntity = entitiesData.find((entity) => entity.id === actorId);
+
+  // Extract entity data from the matching entity's documentsData if available
+  const entityDetailsData = matchingEntity?.entity ?? {};
+
+  let mappedProps = {};
+
+  if (entityDetailsData.name) {
+    mappedProps = set('label', entityDetailsData.name)(mappedProps);
```

Comment:
Can you elaborate what's the difference between 
`mappedProps.label = entityDetailsData.name;`

to 

`mappedProps = set('label', entityDetailsData.name)(mappedProps);`
?


I'm familiar with set when you wish to set 2+ level property of an object. But I'm not sure if there is any benefit from top level fields set